packages/roosterjs-editor-plugins/lib/plugins/Paste/imageConverter/convertPastedImage.ts

```diff
@@ -0,0 +1,30 @@
+import { BeforePasteEvent, TrustedHTMLHandler } from 'roosterjs-editor-types';
+import { getTagOfNode, moveChildNodes } from 'roosterjs-editor-dom';
+
+/**
+ * @internal
+ * Convert pasted content if there are HTML and Image data in the Clipboard
+ * @param event The BeforePaste event
+ */
+export default function convertPastedImage(
+    event: BeforePasteEvent,
+    trustedHTMLHandler: TrustedHTMLHandler
+) {
+    const { fragment, clipboardData } = event;
+
+    if (clipboardData.html && clipboardData.image) {
+        //If there are Html in the clipboard, and the html body only have one img children, use the HTML
+        const doc = new DOMParser().parseFromString(
```

Comment:
This means we need to parse the HTML content again in a very comment case, but in most case after parsing we will see that there will not only be an IMAGE, so we should do nothing. That means this check will be wasted in most time.

Consider add some hint info in the ClipboardData object so that when createPasteFragment we can set the hint info there since we have the parsed doc at the time. Then here we can read it and when it meet some condition, we do this work here. That can save more than 99% parsing cases.